Aur builds fails on arch and manjaro

### Operating System

Linux

### Version of Prism Launcher

ca4b58d5d7ecf0ba13d410ecb177d1aef85b7e26

### Version of Qt

6.6.0

### Description of bug

The build on arch system is broken

### Steps to reproduce

Just try to install/update [prismlauncher-git](https://aur.archlinux.org/packages/prismlauncher-git) from aur

### Suspected cause

<details>
  <summary>Build log</summary>
  
```
$cmake --build build && gdb -ex run ./build/prismlauncher
[0/1] Re-running CMake...
-- Not enabling IPO / LTO on debug builds
-- Git commit: ca4b58d5d7ecf0ba13d410ecb177d1aef85b7e26
-- Git tag: -128-NOTFOUND
-- Git refspec: refs/heads/arch_build
Installing in /home/user/Projects/PrismLauncher/install. Run /home/user/Projects/PrismLauncher/build/prefix.sh to set the environment for Launcher.
-- Using system zlib
-- Using system QuaZip
-- Using system tomlplusplus
-- Using system cmark
-- Using system ghc_filesystem
-- Configuring done (0.8s)
-- Generating done (0.1s)
-- Build files have been written to: /home/user/Projects/PrismLauncher/build
[5/28] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/translations/TranslationsModel.cpp.o
FAILED: launcher/CMakeFiles/Launcher_logic.dir/translations/TranslationsModel.cpp.o 
/usr/bin/c++ -DLAUNCHER_APPLICATION -DLAUNCHER_TEST -DQT_CONCURRENT_LIB -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -I/home/user/Projects/PrismLauncher/build/launcher -I/home/user/Projects/PrismLauncher/launcher -I/home/user/Projects/PrismLauncher/build/launcher/Launcher_logic_autogen/include -I/home/user/Projects/PrismLauncher/libraries/systeminfo/include -I/home/user/Projects/PrismLauncher/libraries/murmur2 -I/home/user/Projects/PrismLauncher/libraries/murmur2/src -I/home/user/Projects/PrismLauncher/libraries/libnbtplusplus/include -I/home/user/Projects/PrismLauncher/build/libraries/libnbtplusplus -I/home/user/Projects/PrismLauncher/libraries/qdcss/include -I/home/user/Projects/PrismLauncher/buildconfig -I/home/user/Projects/PrismLauncher/libraries/katabasis/include -I/home/user/Projects/PrismLauncher/libraries/gamemode/include -I/home/user/Projects/PrismLauncher/libraries/LocalPeer/include -I/home/user/Projects/PrismLauncher/libraries/rainbow/include -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtCore5Compat -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/QuaZip-Qt6-1.4 -isystem /usr/include/QuaZip-Qt6-1.4/quazip -Wall -pedantic -fstack-protector-strong --param=ssp-buffer-size=4  -DQT_NO_DEPRECATED_WARNINGS=Y -DQT_DISABLE_DEPRECATED_BEFORE=0x050C00 -DTOML_ENABLE_FLOAT16=0 -g -std=gnu++17 -mno-direct-extern-access -DTOML_HEADER_ONLY=0 -DTOML_SHARED_LIB=1 -MD -MT launcher/CMakeFiles/Launcher_logic.dir/translations/TranslationsModel.cpp.o -MF launcher/CMakeFiles/Launcher_logic.dir/translations/TranslationsModel.cpp.o.d -o launcher/CMakeFiles/Launcher_logic.dir/translations/TranslationsModel.cpp.o -c /home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp: In member function ‘Language* TranslationsModel::findLanguage(const QString&)’:
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:426:16: error: cannot convert ‘QList<Language>::iterator’ to ‘Language*’ in return
  426 |         return found;
      |                ^~~~~
      |                |
      |                QList<Language>::iterator
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp: In member function ‘QModelIndex TranslationsModel::selectedIndex()’:
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:532:28: error: no match for ‘operator-’ (operand types are ‘Language*’ and ‘QList<Language>::iterator’)
  532 |         return index(found - d->m_languages.begin(), 0, QModelIndex());
      |                      ~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~
      |                      |                           |
      |                      Language*                   QList<Language>::iterator
In file included from /usr/include/qt6/QtCore/qsharedpointer.h:12,
                 from /usr/include/qt6/QtCore/qdebug.h:16,
                 from /usr/include/qt6/QtCore/qvariant.h:11,
                 from /usr/include/qt6/QtCore/qabstractitemmodel.h:11,
                 from /usr/include/qt6/QtCore/QAbstractListModel:1,
                 from /home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.h:18,
                 from /home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:37:
/usr/include/qt6/QtCore/qsharedpointer_impl.h:749:63: note: candidate: ‘template<class T, class X> typename QSharedPointer<T>::difference_type operator-(const QSharedPointer<T>&, const QSharedPointer<T>&)’
  749 | Q_INLINE_TEMPLATE typename QSharedPointer<T>::difference_type operator-(const QSharedPointer<T> &ptr1, const QSharedPointer<X> &ptr2)
      |                                                               ^~~~~~~~
/usr/include/qt6/QtCore/qsharedpointer_impl.h:749:63: note:   template argument deduction/substitution failed:
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:532:51: note:   mismatched types ‘const QSharedPointer<T>’ and ‘Language*’
  532 |         return index(found - d->m_languages.begin(), 0, QModelIndex());
      |                                                   ^
/usr/include/qt6/QtCore/qsharedpointer_impl.h:754:63: note: candidate: ‘template<class T, class X> typename QSharedPointer<T>::difference_type operator-(const QSharedPointer<T>&, X*)’
  754 | Q_INLINE_TEMPLATE typename QSharedPointer<T>::difference_type operator-(const QSharedPointer<T> &ptr1, X *ptr2)
      |                                                               ^~~~~~~~
/usr/include/qt6/QtCore/qsharedpointer_impl.h:754:63: note:   template argument deduction/substitution failed:
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:532:51: note:   mismatched types ‘const QSharedPointer<T>’ and ‘Language*’
  532 |         return index(found - d->m_languages.begin(), 0, QModelIndex());
      |                                                   ^
/usr/include/qt6/QtCore/qsharedpointer_impl.h:759:63: note: candidate: ‘template<class T, class X> typename QSharedPointer<T>::difference_type operator-(T*, const QSharedPointer<T>&)’
  759 | Q_INLINE_TEMPLATE typename QSharedPointer<X>::difference_type operator-(T *ptr1, const QSharedPointer<X> &ptr2)
      |                                                               ^~~~~~~~
/usr/include/qt6/QtCore/qsharedpointer_impl.h:759:63: note:   template argument deduction/substitution failed:
/home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:532:51: note:   ‘QList<Language>::iterator’ is not derived from ‘const QSharedPointer<T>’
  532 |         return index(found - d->m_languages.begin(), 0, QModelIndex());
      |                                                   ^
In file included from /usr/include/qt6/QtCore/QtCore:148,
                 from /usr/include/qt6/QtNetwork/QtNetworkDepends:3,
                 from /usr/include/qt6/QtNetwork/QtNetwork:3,
                 from /home/user/Projects/PrismLauncher/launcher/net/NetJob.h:39,
                 from /home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:51:
/usr/include/qt6/QtCore/qmargins.h:141:27: note: candidate: ‘constexpr QMargins operator-(const QMargins&, const QMargins&)’
  141 | constexpr inline QMargins operator-(const QMargins &m1, const QMargins &m2) noexcept
      |                           ^~~~~~~~
/usr/include/qt6/QtCore/qmargins.h:141:53: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QMargins&’
  141 | constexpr inline QMargins operator-(const QMargins &m1, const QMargins &m2) noexcept
      |                                     ~~~~~~~~~~~~~~~~^~
/usr/include/qt6/QtCore/qmargins.h:159:27: note: candidate: ‘constexpr QMargins operator-(const QMargins&, int)’
  159 | constexpr inline QMargins operator-(const QMargins &lhs, int rhs) noexcept
      |                           ^~~~~~~~
/usr/include/qt6/QtCore/qmargins.h:159:53: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QMargins&’
  159 | constexpr inline QMargins operator-(const QMargins &lhs, int rhs) noexcept
      |                                     ~~~~~~~~~~~~~~~~^~~
/usr/include/qt6/QtCore/qmargins.h:260:27: note: candidate: ‘constexpr QMargins operator-(const QMargins&)’
  260 | constexpr inline QMargins operator-(const QMargins &margins) noexcept
      |                           ^~~~~~~~
/usr/include/qt6/QtCore/qmargins.h:260:27: note:   candidate expects 1 argument, 2 provided
/usr/include/qt6/QtCore/qmargins.h:395:28: note: candidate: ‘constexpr QMarginsF operator-(const QMarginsF&, const QMarginsF&)’
  395 | constexpr inline QMarginsF operator-(const QMarginsF &lhs, const QMarginsF &rhs) noexcept
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qmargins.h:395:55: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QMarginsF&’
  395 | constexpr inline QMarginsF operator-(const QMarginsF &lhs, const QMarginsF &rhs) noexcept
      |                                      ~~~~~~~~~~~~~~~~~^~~
/usr/include/qt6/QtCore/qmargins.h:413:28: note: candidate: ‘constexpr QMarginsF operator-(const QMarginsF&, qreal)’
  413 | constexpr inline QMarginsF operator-(const QMarginsF &lhs, qreal rhs) noexcept
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qmargins.h:413:55: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QMarginsF&’
  413 | constexpr inline QMarginsF operator-(const QMarginsF &lhs, qreal rhs) noexcept
      |                                      ~~~~~~~~~~~~~~~~~^~~
/usr/include/qt6/QtCore/qmargins.h:487:28: note: candidate: ‘constexpr QMarginsF operator-(const QMarginsF&)’
  487 | constexpr inline QMarginsF operator-(const QMarginsF &margins) noexcept
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qmargins.h:487:28: note:   candidate expects 1 argument, 2 provided
In file included from /usr/include/qt6/QtCore/QtCore:200:
/usr/include/qt6/QtCore/qrect.h:441:24: note: candidate: ‘constexpr QRect operator-(const QRect&, const QMargins&)’
  441 | constexpr inline QRect operator-(const QRect &lhs, const QMargins &rhs) noexcept
      |                        ^~~~~~~~
/usr/include/qt6/QtCore/qrect.h:441:47: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QRect&’
  441 | constexpr inline QRect operator-(const QRect &lhs, const QMargins &rhs) noexcept
      |                                  ~~~~~~~~~~~~~^~~
/usr/include/qt6/QtCore/qrect.h:869:25: note: candidate: ‘constexpr QRectF operator-(const QRectF&, const QMarginsF&)’
  869 | constexpr inline QRectF operator-(const QRectF &lhs, const QMarginsF &rhs) noexcept
      |                         ^~~~~~~~
/usr/include/qt6/QtCore/qrect.h:869:49: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QRectF&’
  869 | constexpr inline QRectF operator-(const QRectF &lhs, const QMarginsF &rhs) noexcept
      |                                   ~~~~~~~~~~~~~~^~~
In file included from /usr/include/qt6/QtGui/qimage.h:12,
                 from /usr/include/qt6/QtGui/qpixmap.h:13,
                 from /usr/include/qt6/QtGui/qbitmap.h:8,
                 from /usr/include/qt6/QtGui/qcursor.h:10,
                 from /usr/include/qt6/QtWidgets/qapplication.h:12,
                 from /usr/include/qt6/QtWidgets/QApplication:1,
                 from /home/user/Projects/PrismLauncher/launcher/Application.h:40,
                 from /home/user/Projects/PrismLauncher/launcher/translations/TranslationsModel.cpp:55:
/usr/include/qt6/QtGui/qtransform.h:353:19: note: candidate: ‘QTransform operator-(const QTransform&, qreal)’
  353 | inline QTransform operator -(const QTransform &a, qreal n)
      |                   ^~~~~~~~
/usr/include/qt6/QtGui/qtransform.h:353:48: note:   no known conversion for argument 1 from ‘Language*’ to ‘const QTransform&’
  353 | inline QTransform operator -(const QTransform &a, qreal n)
      |                              ~~~~~~~~~~~~~~~~~~^
[6/28] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/modplatform/modrinth/ModrinthPackManifest.cpp.o
ninja: build stopped: subcommand failed.
```
  
</details>

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

--- a/launcher/translations/TranslationsModel.cpp
+++ b/launcher/translations/TranslationsModel.cpp
@@ -206,7 +206,7 @@ void TranslationsModel::indexReceived()
         reloadLocalFiles();
 
         auto language = d->m_system_locale;
-        if (!findLanguage(language)) {
+        if (!findLanguageAsOptional(language).has_value()) {
             language = d->m_system_language;
         }
         selectLanguage(language);
@@ -417,14 +417,17 @@ int TranslationsModel::columnCount([[maybe_unused]] const QModelIndex& parent) c
     return 2;
 }
 
-Language* TranslationsModel::findLanguage(const QString& key)
+QVector<Language>::Iterator TranslationsModel::findLanguage(const QString& key)
 {
-    auto found = std::find_if(d->m_languages.begin(), d->m_languages.end(), [&](Language& lang) { return lang.key == key; });
-    if (found == d->m_languages.end()) {
-        return nullptr;
-    } else {
-        return found;
-    }
+    return std::find_if(d->m_languages.begin(), d->m_languages.end(), [&](Language& lang) { return lang.key == key; });
+}
+
+std::optional<Language> TranslationsModel::findLanguageAsOptional(const QString& key)
+{
+    auto found = findLanguage(key);
+    if (found != d->m_languages.end())
+        return *found;
+    return {};
 }
 
 void TranslationsModel::setUseSystemLocale(bool useSystemLocale)
@@ -436,13 +439,13 @@ void TranslationsModel::setUseSystemLocale(bool useSystemLocale)
 bool TranslationsModel::selectLanguage(QString key)
 {
     QString& langCode = key;
-    auto langPtr = findLanguage(key);
+    auto langPtr = findLanguageAsOptional(key);
 
     if (langCode.isEmpty()) {
         d->no_language_set = true;
     }
 
-    if (!langPtr) {
+    if (!langPtr.has_value()) {
         qWarning() << "Selected invalid language" << key << ", defaulting to" << defaultLangCode;
         langCode = defaultLangCode;
     } else {
@@ -527,9 +530,8 @@ bool TranslationsModel::selectLanguage(QString key)
 QModelIndex TranslationsModel::selectedIndex()
 {
     auto found = findLanguage(d->m_selectedLanguage);
-    if (found) {
-        // QVector iterator freely converts to pointer to contained type
-        return index(found - d->m_languages.begin(), 0, QModelIndex());
+    if (found != d->m_languages.end()) {
+        return index(std::distance(d->m_languages.begin(), found), 0, QModelIndex());
     }
     return QModelIndex();
 }
@@ -562,8 +564,8 @@ void TranslationsModel::updateLanguage(QString key)
         qWarning() << "Cannot update builtin language" << key;
         return;
     }
-    auto found = findLanguage(key);
-    if (!found) {
+    auto found = findLanguageAsOptional(key);
+    if (!found.has_value()) {
         qWarning() << "Cannot update invalid language" << key;
         return;
     }
@@ -578,8 +580,8 @@ void TranslationsModel::downloadTranslation(QString key)
         d->m_nextDownload = key;
         return;
     }
-    auto lang = findLanguage(key);
-    if (!lang) {
+    auto lang = findLanguageAsOptional(key);
+    if (!lang.has_value()) {
         qWarning() << "Will not download an unknown translation" << key;
         return;
     }

--- a/launcher/translations/TranslationsModel.h
+++ b/launcher/translations/TranslationsModel.h
@@ -17,6 +17,7 @@
 
 #include <QAbstractListModel>
 #include <memory>
+#include <optional>
 
 struct Language;
 
@@ -40,7 +41,8 @@ class TranslationsModel : public QAbstractListModel {
     void setUseSystemLocale(bool useSystemLocale);
 
    private:
-    Language* findLanguage(const QString& key);
+    QVector<Language>::Iterator findLanguage(const QString& key);
+    std::optional<Language> findLanguageAsOptional(const QString& key);
     void reloadLocalFiles();
     void downloadTranslation(QString key);
     void downloadNext();

